samza-azure/src/main/java/org/apache/samza/AzureLeaderElector.java

```diff
@@ -0,0 +1,118 @@
+ */
+
+package org.apache.samza;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.samza.coordinator.LeaderElector;
+import org.apache.samza.coordinator.LeaderElectorListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Class to facilitate leader election in Azure.
+ * The processor that acquires the lease on the blob becomes the leader.
+ * The lease ID is null initially. It is generated by Azure when the processor acquires the lease, and updated accordingly.
+ * Every processor requires a valid active lease ID in order to perform successful write and delete operations on the blob.
+ * Read operations from the blob are not dependent on the lease ID.
+ */
+public class AzureLeaderElector implements LeaderElector {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AzureLeaderElector.class);
+  private static final int LEASE_TIME_IN_SEC = 60;
+  private final LeaseBlobManager leaseBlobManager;
+  private LeaderElectorListener leaderElectorListener = null;
+  private AtomicReference<String> leaseId;
+  private AtomicBoolean isLeader;
+  private ScheduledFuture renewLeaseSF;
+  private ScheduledFuture livenessSF;
+
+  public AzureLeaderElector(LeaseBlobManager leaseBlobManager) {
+    this.isLeader = new AtomicBoolean(false);
+    this.leaseBlobManager = leaseBlobManager;
+    this.isLeader = new AtomicBoolean(false);
+    this.leaseId = new AtomicReference<>(null);
+  }
+
+  @Override
+  public void setLeaderElectorListener(LeaderElectorListener listener) {
+    this.leaderElectorListener = listener;
+  }
+
+  /**
+   * Tries to become the leader by acquiring a lease on the blob.
+   * Invokes the listener on becoming the leader.
+   */
+  @Override
+  public void tryBecomeLeader() {
+    leaseId.getAndSet(leaseBlobManager.acquireLease(LEASE_TIME_IN_SEC, leaseId.get()));
+    if (leaseId.get() != null) {
+      LOG.info("Became leader!");
+      isLeader.set(true);
+      leaderElectorListener.onBecomingLeader();
+    }
+  }
+
+  /**
+   * Releases the lease in order to resign leadership. It also stops all schedulers scheduled by the leader.
+   */
+  @Override
+  public void resignLeadership() {
+    if (isLeader.get()) {
+      boolean status = leaseBlobManager.releaseLease(leaseId.get());
+      if (status) {
+        isLeader.set(false);
+        leaseId = null;
+        renewLeaseSF.cancel(true);
+        livenessSF.cancel(true);
+      }
+    } else {
+      LOG.info("Can't release the lease because it is not the leader and does not hold an active lease.");
+    }
+  }
+
+  /**
+   * Checks whether it's a leader
+   * @return true if it is the leader, false otherwise
+   */
+  @Override
+  public boolean amILeader() {
+    return isLeader.get();
+  }
+
+  public String getLeaseId() {
+    return leaseId.get();
+  }
+
+  public void setRenewLeaseScheduledFuture(ScheduledFuture sf) {
```

Comment:
I am curious to see how this `ScheduledFuture` is used. Until then, can you please add a line or two of docs explaining what `livenessSF` and `renewLeaseSF` is meant for?